packages/core-js/modules/web.structured-clone.js

```diff
@@ -0,0 +1,454 @@
+var stringValueOf = uncurryThis(''.valueOf);
+var getTime = uncurryThis(Date.prototype.getTime);
+var PERFORMANCE_MARK = uid('structuredClone');
+var DATA_CLONE_ERROR = 'DataCloneError';
+var TRANSFERRING = 'Transferring';
+
+// waiting for https://github.com/zloirock/core-js/pull/991
+var DOMException = function (message, name) {
+  try {
+    return new global.DOMException(message, name);
+  } catch (error) {
+    return TypeError(message);
+  }
+};
+
+var checkBasicSemantic = function (structuredCloneImplementation) {
+  return !fails(function () {
+    var set1 = new global.Set([7]);
+    var set2 = structuredCloneImplementation(set1);
+    var number = structuredCloneImplementation(Object(7));
+    return set2 == set1 || !set2.has(7) || typeof number != 'object' || number != 7;
+  });
+};
+
+// https://github.com/whatwg/html/pull/5749
+var checkNewErrorsSemantic = function (structuredCloneImplementation) {
+  return !fails(function () {
+    var test = structuredCloneImplementation(new global.AggregateError([1], PERFORMANCE_MARK, { cause: 3 }));
+    return test.name != 'AggregateError' || test.errors[0] != 1 || test.message != PERFORMANCE_MARK || test.cause != 3;
+  });
+};
+
+// FF94+, Safari TP134+, Chrome Canary 98+, NodeJS 17.0+, Deno 1.14+
+var nativeStructuredClone = global.structuredClone;
+
+// Chrome 78+, Safari 14.1+, NodeJS 16.0+, Deno 1.11+ (old Deno implementations too naive)
+var structuredCloneFromMark = !nativeStructuredClone && (function (fromMarkConstructor, fromMark) {
+  return checkBasicSemantic(fromMarkConstructor) ? fromMarkConstructor
+    : checkBasicSemantic(fromMark) && fromMark;
+})(function (value) {
+  return new PerformanceMark(PERFORMANCE_MARK, { detail: value }).detail;
+}, function (value) {
+  // NodeJS haven't `PerformanceMark` constructor
+  var result = call(mark, performance, PERFORMANCE_MARK, { detail: value });
+  call(clearMarks, performance, PERFORMANCE_MARK);
+  return result.detail;
+});
+
+var nativeRestrictedStructuredClone = checkBasicSemantic(nativeStructuredClone) ? nativeStructuredClone : structuredCloneFromMark;
+
+// Chrome 82- implementation swaps `.name` and `.message` of cloned `DOMException`
+// current Safari implementation can't clone errors
+// no one of current implementations supports new (html/5749) error cloning semantic
+var USE_STRUCTURED_CLONE_FROM_MARK = !IS_PURE && !nativeStructuredClone && checkNewErrorsSemantic(structuredCloneFromMark);
+
+var throwUncloneable = function (type) {
+  throw new DOMException('Uncloneable type: ' + type, DATA_CLONE_ERROR);
+};
+
+var throwUnpolyfillable = function (type, kind) {
+  throw new DOMException((kind || 'Cloning') + ' of ' + type + ' cannot be properly polyfilled in this engine', DATA_CLONE_ERROR);
+};
+
+var structuredCloneInternal = function (value, map) {
+  if (isSymbol(value)) throwUncloneable('Symbol');
+  if (!isObject(value)) return value;
+  if (USE_STRUCTURED_CLONE_FROM_MARK) return structuredCloneFromMark(value);
```

Comment:
You forgot the check if map was given or not :)

```suggestion
  if (USE_STRUCTURED_CLONE_FROM_MARK && !map) return structuredCloneFromMark(value);
```

Reply:
Thanks. However, most likely, `USE_STRUCTURED_CLONE_FROM_MARK` will be completely removed since native `structuredClone` is already available in unstable versions of all modern engines, but no one of them still not passes new errors cloning tests.

Reply:
I agree. The only possible scenario I can think of is that Node.js v16 will backport `AggregateError` support as a `MessagePort` bugfix, but that's unlikely too.